### PR TITLE
Change line numbers in documentation

### PR DESCRIPTION
**Pull request by mekanix**
I changed all occurrences of `Meta` that I knew how to, as well. Line numbers change when I replace `Meta` with `ormar_config`, so I did that. If you grep, you'll find more `Meta` instances around, but I don't know how to change them. For example, I never worked with signals, so I don't know how's that implemented now.

### CODE REVIEW

**Comment by mekanix** [comment]
Do you want to teach me how new signals work, or you want to fill in that info yourself?

**Comment by collerek** [comment]
@mekanix The signals are working like they used to, not sure what you refer to regarding Meta in there?

**Comment by mekanix** [comment]
For example [here](https://github.com/mekanix/ormar/blob/line-numbers/docs/signals.md?plain=1#L109)

**Comment by collerek** [comment]
Ah, ok, so you can assume what was in `Meta` will be now in `ormar_config`, it's registered [here](https://github.com/collerek/ormar/blob/2d798666c01a52012a1d6d754e5ae2b89acefb8f/ormar/models/metaclass.py#L121)

**Comment by mekanix** [comment]
The only thing left is [this mention of Config](https://github.com/mekanix/ormar/blob/line-numbers/docs/models/index.md?plain=1#L320) that I don't know what to do with. Can you help, please?

File: docs/models/index.md [hunk 1]
```diff
@@ -348,69 +342,59 @@ If you try to do so the `ModelError` will be raised.
...
+    database=databases.Database(DATABASE_URL),
+    metadata=sqlalchemy.MetaData(),
+)
 
 # now default sort by name descending
 class Author(ormar.Model):
-    class Meta(BaseMeta):
-        tablename = "authors"
-        orders_by = ["-name"]
+    ormar_config = base_ormar_config.copy(orders_by = ["-name"])
```
Comment (collerek) [inline docs/models/index.md:397]: would add back explicit table name as this is one of the first section users read.

File: docs/models/index.md [hunk 2]
```diff
@@ -170,12 +174,7 @@ class MyQuerySetClass(QuerySet):
 
         
 class Book(ormar.Model):
-    
-    class Meta(ormar.ModelMeta):
-        metadata = metadata
-        database = database
-        tablename = "book"
-        queryset_class = MyQuerySetClass
+    ormar_config = base_ormar_config.copy(queryset_class=MyQuerySetClass)
```
Comment (collerek) [inline docs/models/index.md:177]: table_name was set to `book` but in now books by default

File: docs/fastapi/response.md [hunk 3]
```diff
@@ -111,22 +113,21 @@ One is a dictionary with nested fields that represents the model tree structure,
...
-    database = database
+base_ormar_config = ormar.OrmarConfig(
+    metadata=metadata
+    database=database
+)
 
 class Category(ormar.Model):
-    class Meta(BaseMeta):
-        tablename: str = "categories"
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/fastapi/response.md:122]: Changed table name (default would be `categorys`)

File: docs/fields/common-parameters.md [hunk 4]
```diff
@@ -184,12 +184,15 @@ So it's on you as a user to provide a type that is valid in the context of given
...
+)
+
+
 # sample overwrites
 class OverwriteTest(ormar.Model):
-    class Meta:
-        tablename = "overwrites"
-        metadata = metadata
-        database = database
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/fields/common-parameters.md:195]: Changed table name

File: docs/fields/encryption.md [hunk 5]
```diff
@@ -59,8 +63,7 @@ Note that since this backend never decrypt the stored value it's only applicable
 
 ```python
 class Hash(ormar.Model):
-    class Meta(BaseMeta):
-        tablename = "hashes"
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/fields/encryption.md:66]: changed table name

File: docs/fields/field-types.md [hunk 6]
```diff
@@ -160,11 +160,16 @@ That way you can i.e. set the value by API, even if value is not `utf-8` compati
...
+    database=database
+)
+
+
 class LargeBinaryStr(ormar.Model):
-    class Meta:
-        tablename = "my_str_blobs"
-        metadata = metadata
-        database = database
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/fields/field-types.md:172]: changed table name

File: docs/models/inheritance.md [hunk 7]
```diff
@@ -45,10 +52,7 @@ class DateFieldsMixins:
 
 # a models can inherit from one or more mixins
 class Category(ormar.Model, DateFieldsMixins, AuditMixin):
-    class Meta(ormar.ModelMeta):
-        tablename = "categories"
-        metadata = metadata
-        database = db
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/models/inheritance.md:55]: Changed table name

File: docs/models/inheritance.md [hunk 8]
```diff
@@ -91,28 +95,27 @@ otherwise an error will be raised.
...
 
     created_date: datetime.datetime = ormar.DateTime(default=datetime.datetime.now)
     updated_date: datetime.datetime = ormar.DateTime(default=datetime.datetime.now)
 
 
 # that way you do not have to provide metadata and databases in concrete class
 class Category(DateFieldsModel, AuditModel):
-    class Meta(ormar.ModelMeta):
-        tablename = "categories"
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/models/inheritance.md:118]: changed table name

File: docs/models/inheritance.md [hunk 9]
```diff
@@ -379,16 +366,13 @@ class Car2(ormar.Model):
...
-        tablename = "trucks2"
+    ormar_config = base_ormar_config.copy(tablename="trucks2")
 
     max_capacity: int = ormar.Integer()
 
 
 class Bus2(Car2):
-    class Meta:
-        tablename = "buses2"
+    ormar_config = base_ormar_config.copy(tablename="busses2")
```
Comment (collerek) [inline docs/models/inheritance.md:375]: typo

File: docs/models/inheritance.md [hunk 10]
```diff
@@ -499,10 +481,11 @@ class DateFieldsModel(ormar.Model):
 
 
 class Category(DateFieldsModel, AuditModel):
-    class Meta(ormar.ModelMeta):
-        tablename = "categories"
+    ormar_config = base_ormar_config.copy(
+        tablename="categories",
         # set fields that should be skipped
-        exclude_parent_fields = ["updated_by", "updated_date"]
+        exclude_parent_fields = ["updated_by", "updated_date"],
```
Comment (collerek) [inline docs/models/inheritance.md:487]: spaces around equal sign

File: docs/queries/filter-and-sort.md [hunk 11]
```diff
@@ -823,8 +806,8 @@ Order in which order_by clauses are applied is as follows:
 
   * Explicitly passed `order_by()` calls in query
   * Relation passed `orders_by` and `related_orders_by` if exists
-  * Model `Meta` class `orders_by`
-  * Model `primary_key` column ascending (fallback, used if none of above provided)
+  * Model's `ormar_config` class `orders_by`
```
Comment (collerek) [inline docs/queries/filter-and-sort.md:809]: now it's not a class but an instance

File: docs/queries/joins-and-subqueries.md [hunk 12]
```diff
@@ -82,31 +76,22 @@ You can provide a string or a list of strings (or a field/ list of fields)
 
 ```python
 class SchoolClass(ormar.Model):
-    class Meta:
-        tablename = "schoolclasses"
-        metadata = metadata
-        database = database
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/queries/joins-and-subqueries.md:79]: changed table name

File: docs/queries/joins-and-subqueries.md [hunk 13]
```diff
@@ -82,31 +76,22 @@ You can provide a string or a list of strings (or a field/ list of fields)
...
     name: str = ormar.String(max_length=100)
     department: Optional[Department] = ormar.ForeignKey(Department, nullable=False)
 
 
 class Category(ormar.Model):
-    class Meta:
-        tablename = "categories"
-        metadata = metadata
-        database = database
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/queries/joins-and-subqueries.md:87]: changed table name

File: docs/transactions.md [hunk 14]
```diff
@@ -15,35 +15,29 @@ async with database.transaction():
...
     id: int = ormar.Integer(primary_key=True)
     name: str = ormar.String(max_length=255)
-
-# database is accessible from class
-database = Author.Meta.database
-
-# as well as from instance
-author = Author(name="Stephen King")
-database = author.Meta.database
-
```
Comment (collerek) [inline docs/transactions.md:46]: please restore it, just change Meta to ormar_config like everywhere else

**Review by collerek — CHANGES_REQUESTED** [review]
hi! In general looks good, great job! Some minor remarks (mostly related to missing table names in ormar_configs)

File: README.md [hunk 15]
```diff
@@ -215,10 +207,9 @@ class Book(ormar.Model):
 # create the database
 # note that in production you should use migrations
 # note that this is not required if you connect to existing database
-engine = sqlalchemy.create_engine(DATABASE_URL)
 # just to be sure we clear the db before
-metadata.drop_all(engine)
-metadata.create_all(engine)
+base_ormar_config.metadata.drop_all(engine)
+base_ormar_config.metadata.create_all(engine)
```
[the remark this answers is not in the document]
Comment (mekanix) [reply README.md]: What does it mean to run readme? Can I do it in the browser or you mean copy/paste the code and check it?

[the remark this answers is not in the document]
Comment (collerek) [reply README.md]: Ah sorry, forgot to explain, there is a file `examples/script_from_readme.py` that should 1:1 match what is in the readme (might need changes, not sure if it was already updated in v2 pr). That one you could run and copy paste to readme after changes so they match. Unfortunately, github/pypi readme files do not support the embedding of code as we have in docs.

[the remark this answers is not in the document]
Comment (mekanix) [reply README.md]: I totally missed this comment. I'll work on this and post and update in few days (sorry for not doing it sooner, I'm very busy this week).

[the remark this answers is not in the document]
Comment (collerek) [reply README.md]: No worries, take your time I'm quite busy too. Note also other comments from the review about changed table names

[the remark this answers is not in the document]
Comment (mekanix) [reply README.md]: I think everything is good now.

File: docs/fastapi/index.md [hunk 16]
```diff
@@ -72,21 +69,21 @@ Define ormar models with appropriate fields.
...
+    database = database
+)
+
+
 class Category(ormar.Model):
-    class Meta:
-        tablename = "categories"
-        metadata = metadata
-        database = database
+    ormar_config = base_ormar_config.copy()
```
Comment (collerek) [inline docs/fastapi/index.md:79]: missed changed name here

File: docs/fields/encryption.md (hunk 5 above)
Comment (collerek) [reply docs/fields/encryption.md:66]: you didn't fix this, should be `ormar_config = base_ormar_config.copy(tablename="hashes")`

**Review by collerek — CHANGES_REQUESTED** [review]
There are still not addressed table name changes

**Comment by collerek** [comment]
Some of the old comments are not addressed too (also names changes)

**Comment by mekanix** [comment]
My knee is killing me so my focus is bad. Sorry you have to go through the PR so many times. Well, get there :smile:

**Comment by collerek** [comment]
Hi, do you know when you will be able to provide those few final changes?

**Comment by mekanix** [comment]
I went through the `git show` and checked `tablename`. I caught few more and I squashed into one commit. My knee is hurt which means I have time, but the pain does steal the focus, so some errors might have sneaked in.

**Review by collerek — APPROVED** [review]

**Pull request merged**